src/pepr/operator/controllers/istio/ambient-waypoint.spec.ts

```
@@ -0,0 +1,616 @@
+          name: "test-sso",
+        },
+      ]),
+      expected: false,
+    },
+    {
+      name: "should return false when package spec is undefined",
+      pkg: { metadata: { name: "test", namespace: "test" } } as UDSPackage,
+      expected: false,
+    },
+    {
+      name: "should return false when package is undefined",
+      pkg: undefined as unknown as UDSPackage,
+      expected: false,
+    },
+  ];
+
+  test.each(testCases)("$name", ({ pkg, expected }) => {
+    expect(hasAuthserviceSSO(pkg)).toBe(expected);
+  });
+});
+
+describe("shouldUseAmbientWaypoint", () => {
+  const testCases = [
+    {
+      name: "should return true when in ambient mode with SSO",
+      pkg: createMockPackage("test", { "app.kubernetes.io/name": "test-app" }),
+      expected: true,
+    },
+    {
+      name: "should return false when not in ambient mode",
+      pkg: createMockPackage("test", { "app.kubernetes.io/name": "test-app" }, "sidecar"),
+      expected: false,
+    },
+    {
+      name: "should return false when no serviceMesh config exists",
+      pkg: {
+        metadata: { name: "test", namespace: "test" },
+        spec: {
+          sso: [
+            {
+              clientId: "test-client",
+              name: "test-sso",
+              enableAuthserviceSelector: { "app.kubernetes.io/name": "test-app" },
+            },
+          ],
+        },
+      } as UDSPackage,
+      expected: false,
+    },
+    {
+      name: "should return false when package is undefined",
+      pkg: undefined as unknown as UDSPackage,
+      expected: false,
+    },
+  ];
+
+  test.each(testCases)("$name", ({ pkg, expected }) => {
+    expect(shouldUseAmbientWaypoint(pkg)).toBe(expected);
+  });
+});
+
+describe("getWaypointName", () => {
+  const testCases = [
+    { input: "test", expected: "uds-core-test-waypoint" },
+    { input: "uds-core-test", expected: "uds-core-test-waypoint" },
+    { input: "", expected: "uds-core--waypoint" },
```

q- is there a case where we support a `""` id? or should we throw an error for this?

---

good eyes, ai generated tests FTW. I refactored this code so that it doesn't allow for empty string or even `"   "` space filled strings, and throws an error.